### PR TITLE
Fix latex circuit drawer handling of circuits without gates

### LINKED ISSUE #5025: circuit without gates cannot be drew on latex
### Information

- **Qiskit Terra version**: master
- **Python version**: 3.7
- **Operating system**: MacOS

### Steps to reproduce the problem
```python
from qiskit import *
qc = QuantumCircuit(2)
qc.draw('latex')
```
```
IndexError: list index out of range
```

### What is the expected behavior?

Same as other drawers:
```python 
qc.draw('text')
```
```
q_0: 
     
q_1: 
```

```python
qc.draw('mpl')
```
image

### PR DESCRIPTION
**Pull request by mlarocca**
### Summary

Fixes #5025 by checking that `self.ops` list is not empty, before checking its first element.

### Details and comments

The issue seems to be that `self.ops` is empty, but its first element is checked in `if self.cregbundle and self.ops[0][0].name == "measure"`
https://github.com/Qiskit/qiskit-terra/blob/master/qiskit/visualization/latex.py#L302

Now, if I have understood correctly, `self.ops` is the list of layers in the circuit.
I noticed that in text.py this list is handled differently (called `instructions` there, but still, if I'm not mistaken...), so my guess is that it's OK for `self.ops` to be empty, and it's fine just checking it before accessing its first element.

**Question**: how come test_visualization.py (TestLatexSourceGenerator) wasn't failing earlier? There are tests about latex conversion...
**Answered**:  this fails only on empty circuits, adding a single gate is enough to avoid the error. 
In the tests, all circuits are non-empty: I added a test for an empty circuit, and verified it was failing before the change, and not anymore afterwards.
(Note: I ran style, lint and tests locally - there are 9 test failing, but they all seem unrelated to this issue).

### CODE REVIEW

File: qiskit/visualization/latex.py [hunk 1]
```diff
@@ -299,7 +299,8 @@ def _get_image_depth(self):
         columns = 2
 
         # add extra column if needed
-        if self.cregbundle and (self.ops[0][0].name == "measure" or self.ops[0][0].condition):
+        if self.cregbundle and (self.ops and self.ops[0] and
```
Comment (mlarocca) [inline qiskit/visualization/latex.py:302]: Three questions here: 
- Can any list of ops (like self.ops[0]) be empty? Should we keep the code defensive and and check it, or can we relax it and just check that self.ops isn't empty?
- Can the list of lists, or any of the lists ever be `None`?
- If it's not possible*, or for the sake of readability, would you prefer an explicit check using `len(self.ops)` etc...?

(*) in that case, I guess it's better to fail spectacularly if `self.ops` or `self.ops[0]` are not lists...

File: test/python/visualization/test_visualization.py [hunk 2]
```diff
@@ -30,6 +30,17 @@
 class TestLatexSourceGenerator(QiskitTestCase):
     """Qiskit latex source generator tests."""
 
+    def test_empty_circuit(self):
+        """Test draw an empty circuit (edge case)."""
```
Comment (1ucian0) [inline test/python/visualization/test_visualization.py:34]:
```suggestion
        """Test draw an empty circuit"""
```

**Review by 1ucian0 — APPROVED** [review]
Thanks a lot for the fix! About the tests... I have to take another look. Probably in a coming PR.

**Comment by mlarocca** [comment]
> Thanks a lot for the fix! About the tests... I have to take another look. Probably in a coming PR.

My pleasure! Please feel free to let me know if I can help any further.

**Pull request merged**